pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionActioner.java

```diff
@@ -223,6 +222,13 @@ private void downloadFile(File pkgFile, boolean isPkgUrlProvided, FunctionMetaDa
 
         if (downloadFromHttp) {
             FunctionCommon.downloadFromHttpUrl(pkgLocationPath, tempPkgFile);
+        } else if (workerConfig.isFunctionsWorkerEnablePackageManagement()) {
```

Comment:
We download with the package service at line 232, why need a new branch? And if we download failed, don't we need to throw the exception?